File: kotlin/sum-of-multiples/src/main/kotlin/SumOfMultiples.kt
object SumOfMultiples {

    fun sum(factors: Set<Int>, limit: Int): Int {
        val multiples = mutableSetOf<Int>()
        for (f in factors.minus(0)) {
            for (m in 1..limit/f) {
                with(f * m) {
                    if (this < limit)
                        multiples.add(this)
                }
            }
        }
        return multiples.sum()
    }
}

File: dart/sum-of-multiples/lib/sum_of_multiples.dart
class SumOfMultiples {
  int sum(List<int> factors, int limit) {
    var multiples = Set<int>();

    void addMultiples(int factor) {
      var multiple = factor;
      while (multiple < limit) {
        multiples.add(multiple);
        multiple += factor;
      }
    }

    factors.where((factor) => factor > 0).forEach(addMultiples);
    return multiples.fold(0, (sum, multiple) => sum + multiple);
  }
}
